modules/tinymce/src/core/test/ts/browser/delete/TableDeleteTest.ts

```diff
@@ -60,13 +60,45 @@ UnitTest.asynctest('browser.tinymce.core.delete.TableDeleteTest', (success, fail
         ])),
 
         Logger.t('Range in only one cell should be noop', GeneralSteps.sequence([
-          tinyApis.sSetContent('<table><tbody><tr><td>a</td><td>b</td></tr></tbody></table>'),
+          tinyApis.sSetContent('<table><tbody><tr><td>ab</td><td>cd</td></tr></tbody></table>'),
           tinyApis.sSetSelection([ 0, 0, 0, 0, 0 ], 0, [ 0, 0, 0, 0, 0 ], 1),
           sDeleteNoop(editor),
-          tinyApis.sAssertContent('<table><tbody><tr><td>a</td><td>b</td></tr></tbody></table>'),
+          tinyApis.sAssertContent('<table><tbody><tr><td>ab</td><td>cd</td></tr></tbody></table>'),
           tinyApis.sAssertSelection([ 0, 0, 0, 0, 0 ], 0, [ 0, 0, 0, 0, 0 ], 1)
         ])),
 
+        Logger.t('All content selected in single cell deletes only content', GeneralSteps.sequence([
+          tinyApis.sSetContent('<table><tbody><tr><td>a</td></tr></tbody></table>'),
+          tinyApis.sSetSelection([ 0, 0, 0, 0, 0 ], 0, [ 0, 0, 0, 0, 0 ], 1),
+          sDelete(editor),
+          tinyApis.sAssertContent('<table><tbody><tr><td>&nbsp;</td></tr></tbody></table>'),
+          tinyApis.sAssertSelection([ 0, 0, 0, 0 ], 0, [ 0, 0, 0, 0 ], 0)
+        ])),
+
+        Logger.t('All content selected in single cell with paragraph deletes only content', GeneralSteps.sequence([
+          tinyApis.sSetContent('<table><tbody><tr><td><p>a</p></td></tr></tbody></table>'),
+          tinyApis.sSetSelection([ 0, 0, 0, 0, 0, 0 ], 0, [ 0, 0, 0, 0, 0, 0 ], 1),
+          sDelete(editor),
+          tinyApis.sAssertContent('<table><tbody><tr><td><p>&nbsp;</p></td></tr></tbody></table>'),
+          tinyApis.sAssertSelection([ 0, 0, 0, 0, 0 ], 0, [ 0, 0, 0, 0, 0 ], 0)
+        ])),
+
+        Logger.t('All content selected in single cell with list deletes only content', GeneralSteps.sequence([
+          tinyApis.sSetContent('<table><tbody><tr><td><ul><li>a</li></ul></td></tr></tbody></table>'),
+          tinyApis.sSetSelection([ 0, 0, 0, 0, 0, 0, 0 ], 0, [ 0, 0, 0, 0, 0 ], 1),
+          sDelete(editor),
+          tinyApis.sAssertContent('<table><tbody><tr><td><ul><li>&nbsp;</li></ul></td></tr></tbody></table>'),
+          tinyApis.sAssertSelection([ 0, 0, 0, 0, 0, 0 ], 0, [ 0, 0, 0, 0, 0, 0 ], 0)
+        ])),
+
+        Logger.t('All content selected in single cell with complex selection deletes only content', GeneralSteps.sequence([
+          tinyApis.sSetContent('<table><tbody><tr><td><p>a</p><ul><li style="list-style-type: none;"><ul><li>b</li></ul></li><li><strong>c</strong></li></ul></td></tr></tbody></table>'),
```

Comment:
Can you please add a case with two divs and paragraphs inside as well? Just want to make sure we've covered all the likely bases.

Reply:
Not sure I follow sorry? As there are many, many ways that can be interpreted. eg: do you mean something like this?
```
<td><div><p>a</p></div><div><p>b</p></div></td>
```
or this:
```
<td><div>a</div><p>b</p></td>
```
or two tests like this:
```
<td><div>a</div><div>b</div></td>
<td><p>a</p><p>b</p></td>
```
or something else?

Reply:
I don't mind adding additional tests, but I'm not going to do ever variant imaginable, so if there's something we're specifically worried about that's what I'd want to add.

Reply:
Sorry, I thought what I said was clear enough. It's the first example; a cell with two div tags, each with a paragraph.

While looking at the `// Clean up any additional leftover nodes` that's the example that struck me as not clearing the contents if `rng.deleteContents()` doesn't do it for us.

I was also concerned about leftover attributes on the block tags but it's arguable whether those need to be removed or not.

Reply:
Right, though that is different again, as now you've mentioned attributes are the concern (which I agree we should have a test for). Anyways, I'll add some more tests to make sure it works as it did natively before and see if that covers the concerns.

Reply:
Added some more tests, so how's that look @TheSpyder?

Reply:
Seems good, and does what I expected it to. If the attributes turn out to be a problem we can deal with that later 🙂